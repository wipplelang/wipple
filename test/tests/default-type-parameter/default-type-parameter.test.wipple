-- [should compile]

X : type (X)
Y : type (Y)

T : A => trait A
instance (T X) : X
instance (T Y) : Y

instance (Show X) : just "X"
instance (Show Y) : just "Y"

foo :: (A : X) where (T A) => A
foo : T

show foo
show (foo :: Y)
