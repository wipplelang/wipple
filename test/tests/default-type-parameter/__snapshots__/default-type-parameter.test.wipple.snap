// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default-type-parameter.test.wipple 1`] = `[]`;

exports[`default-type-parameter.test.wipple 2`] = `
{
  "output": "not implemented
",
}
`;
